shell/src/main/java/alluxio/cli/bundler/CollectInfoAll.java

```diff
@@ -0,0 +1,259 @@
+    return hosts;
+  }
+
+  /**
+   * Main method, starts a new CollectInfoAll shell.
+   * CollectInfoAll will SSH to all hosts and invoke {@link CollectInfo}.
+   * Then collect the tarballs generated on each of the hosts to the localhost.
+   * And tarball all results into one final tarball.
+   *
+   * @param argv array of arguments given by the user's input from the terminal
+   */
+  public static void main(String[] argv) throws IOException {
+    int ret = 0;
+
+    InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+
+    // Execute the Collectors one by one
+    // Reduce the RPC retry max duration to fail earlier for CLIs
+    conf.set(PropertyKey.USER_RPC_RETRY_MAX_DURATION, "5s", Source.DEFAULT);
+    CollectInfoAll shellAll = new CollectInfoAll(conf);
+
+    // Validate commands
+    if (argv.length < 2) {
+      shellAll.printUsage();
+      System.exit(-1);
+    }
+    String targetDir = argv[1];
+
+    List<String> allHosts = new ArrayList<>(shellAll.getHosts());
+    System.out.format("Init thread pool for %s hosts", allHosts.size());
+    shellAll.mExecutor = Executors.newFixedThreadPool(allHosts.size());
+
+    // Invoke infoBundle on each host
+    List<CompletableFuture<CommandReturn>> futureList = new ArrayList<>();
+    for (String host : allHosts) {
+      System.out.format("Execute CollectInfo on host %s", host);
+
+      CompletableFuture<CommandReturn> future = CompletableFuture.supplyAsync(() -> {
+        // We make the assumption that the Alluxio WORK_DIR is the same
+        String workDir = conf.get(PropertyKey.WORK_DIR);
+        String alluxioBinPath = Paths.get(workDir, "bin/alluxio")
+                .toAbsolutePath().toString();
+        System.out.format("host: %s, alluxio path %s", host, alluxioBinPath);
+
+        try {
+          String[] infoBundleArgs =
+                  (String[]) ArrayUtils.addAll(new String[]{alluxioBinPath, "infoBundle"}, argv);
+          CommandReturn cr = ShellUtils.sshExecCommandWithOutput(host, infoBundleArgs);
+          return cr;
+        } catch (Exception e) {
+          LOG.error("Execution failed %s", e);
+          return new CommandReturn(1, e.toString());
+        }
+      }, shellAll.mExecutor);
+      futureList.add(future);
+      System.out.format("Invoked infoBundle command on host %s", host);
+    }
+
+    // Collect the execution results
+    List<CommandReturn> results;
+    try {
+      results = collectAllFuture(futureList).get();
+      System.out.format("Results collected from %d hosts", results.size());
+    } catch (InterruptedException | ExecutionException e) {
+      System.err.format("Failed to collect the results. Error is %s", e.getMessage());
+      LOG.error("Error: %s", e);
+      return;
+    }
+
+    // Inspect all command results
+    if (results.size() != allHosts.size()) {
+      System.out.format(
+              "Error occurred while collection information on the following hosts: "
```

Comment:
My bad, should be "Error occurred while collecting information". Maybe just say "Error occurred while collection information on %d/%d hosts."